src/org/exist/xquery/functions/inspect/ModuleFunctions.java

```diff
@@ -60,15 +60,26 @@ public Sequence eval(Sequence[] args, Sequence contextSequence) throws XPathExce
             Module module = null;
 
             try {
-            if (isCalledAs("module-functions-by-uri"))
-                {module = tempContext.importModule(args[0].getStringValue(), null, null);}
-            else
-                {module = tempContext.importModule(null, null, args[0].getStringValue());}
+                if (isCalledAs("module-functions-by-uri")) {
+                    module = tempContext.importModule(args[0].getStringValue(), null, null);
+                } else {
+                    module = tempContext.importModule(null, null, args[0].getStringValue());
+                }
+                
+            } catch (final XPathException e) {
+                LOG.debug("Failed to import module: " + args[0].getStringValue() + ": " + e.getMessage(), e);
+
+                if (e.getErrorCode().equals(ErrorCodes.XPST0003)) {
```

Comment:
I don't think XPST0003 is the correct error message to use here? It is for static error, but this is at runtime so it must by a dynamic error...